python/python/bystro/ancestry/tests/test_train.py

```diff
@@ -286,3 +288,72 @@ def test_1kgp_vcf_to_dosage():
     for column in processed_vcf.columns:
         for value in processed_vcf[column]:
             assert value in valid_values
+
+
+def test_restrict_loadings_variants_to_vcf():
+    """Tests restriction of gnomad loadings variants to vcf variants."""
+    num_samples = 10
+    num_vars = 10
+    simulated_vcf = generate_simulated_vcf(num_samples, num_vars)
+    sim_vcf_df = convert_sim_vcf_to_df(simulated_vcf)
+    loaded_vcf = convert_1kgp_vcf_to_dosage(sim_vcf_df)
+    processed_sim_vcf = process_vcf_for_pc_transformation(loaded_vcf)
+    # Using same indices, generate sim loadings
+    num_pcs = 30
+    sim_pcs = np.random.random((num_samples, num_pcs))
+    sim_loadings = pd.DataFrame(
+        data=sim_pcs, index=processed_sim_vcf.index, columns=[f"PC{i+1}" for i in range(num_pcs)]
+    )
+    # Run restrict_loadings with sim data
+    pc_loadings_overlap, genos_overlap, num_var_overlap = restrict_loadings_variants_to_vcf(
+        sim_loadings, processed_sim_vcf
+    )
+    # Check for expected columns
+    expected_columns_loadings = ["PC1", "PC2"]
+    expected_columns_genos = ["SampleID1", "SampleID2"]
+    for column in expected_columns_loadings:
+        assert column in pc_loadings_overlap.columns
+    for column in expected_columns_genos:
+        assert column in genos_overlap.columns
+    # Check that the output DataFrames have the expected number of rows
+    expected_num_rows = num_var_overlap
+    assert pc_loadings_overlap.shape[0] == expected_num_rows
```

Comment:
I think this test is a sign that we don't really need num_var_overlap in the return value of this function, because it's redundantly encoding the lengths of these dataframes.

Reply:
I think I wanted to have a readout of num_var_overlap at some point, but wasn't sure where to put it - probably best as part of a log instead though?